mkosi/__init__.py

```diff
@@ -4354,6 +4276,40 @@ def needs_build(config: Union[argparse.Namespace, MkosiConfig]) -> bool:
     return config.verb == Verb.build or (config.verb in MKOSI_COMMANDS_NEED_BUILD and (not config.output.exists() or config.force > 0))
 
 
+def read_subrange(path: Path) -> tuple[int, int]:
+    for line in path.read_text().splitlines():
+        name, start, count = line.split(":")
+
+        if name == pwd.getpwuid(os.getuid()).pw_name:
+            break
+    else:
+        die(f"No mapping found for current user in {path}")
+
+    return int(start), int(count)
+
+
+def become_root() -> None:
+    if os.getuid() == 0:
+        return None
+
+    subuid, subuidcount = read_subrange(Path("/etc/subuid"))
+    subgid, subgidcount = read_subrange(Path("/etc/subgid"))
+
+    event = multiprocessing.Event()
+    pid = os.getpid()
+
+    child = os.fork()
+    if child == 0:
+        event.wait()
+        run(["newuidmap", str(pid), "0", str(os.getuid()), "1", "1", str(subuid), str(int(subuidcount) - 1)])
+        run(["newgidmap", str(pid), "0", str(os.getuid()), "1", "1", str(subgid), str(int(subgidcount) - 1)])
```

Comment:
```suggestion
        run(["newuidmap", f"{pid}, "0", f"{os.getuid()}", "1", "1", f"{subuid}", f"{subuidcount - 1}"])
        run(["newgidmap", f"{pid}", "0", f"{os.getuid()}", "1", "1", f"{subgid}", f"{subgidcount - 1}"])
```
Feels like less obstrusive than the `str` calls, also `subuidcount` and and `subgidcount` are already ints coming out of `read_subrange`.

Also this code might make more sense in `run.py`.

Reply:
Not addressed.

Reply:
To be fair, I disagree here, I think  using `str()` is much nicer than using f strings here

Reply:
Addressed the other thing

Reply:
> To be fair, I disagree here, I think using `str()` is much nicer than using f strings here

Totally fair, this actually was more about the `int` calls.